test/Transforms/memory_banking_attrs.mlir

```diff
@@ -0,0 +1,123 @@
+// RUN: circt-opt %s -memory-banking="factor=3 dimension=1" | FileCheck %s --check-prefix ATTRIBUTE
```

Comment:
Having multiple banking configurations in the same file just complicates everything and I had to make a new test file for it..

Reply:
Makes sense. I will note that this is a rather large test example. Can this be made into a smaller example? I'll leave this to your discretion.

Reply:
I decided to keep this large example just to demonstrate a full working example with different configurations